src/Core/Tags/WebScrap.Core.Tags/TagsProcessorBase.cs

```diff
@@ -48,18 +49,26 @@ public ImmutableArray<ProcessedTag> Process(ReadOnlySpan<char> html)
         UnprocessedTag[] openedTags, 
         ProcessedTag tag) { }
 
-    private int Process(ReadOnlySpan<char> html, int charsProcessed)
+    int Process(ReadOnlySpan<char> html, int charsProcessed)
     {
         var currentHtml = html[charsProcessed..];
-        openedTags.TryPeek(out var lastOpenedTag);
 
+        if (CommentsSkipper.TrySkipComment(currentHtml, out var processed))
```

Comment:
I realize that Skip has became a traditional naming, but it's not late to move from it.